rocks-lib/src/build/mod.rs

```diff
@@ -62,11 +62,13 @@ pub fn build(rockspec: Rockspec, config: &Config) -> Result<()> {
     // operations in the temporary directory itself and then copy all results over once they've
     // succeeded.
 
-    let lua_version = LuaVersion::try_from(&rockspec)?;
+    // TODO(vhyrro): Create a unified way of accessing the Lua version with centralized error
+    // handling.
+    let lua_version = config.lua_version.clone().or(rockspec.lua_version()).ok_or_eyre("lua version not set! Please provide a version through `--lua-version <ver>`")?;
```

Comment:
question: what if the rockspec's lua version is incompatible with the config's lua version? 
Would it be better if the rockspec took precedence here?

Reply:
I think conflicts should result in an error. If a user explicitly requires Lua 5.1, but their project is in conflict, then allowing it could only cause problems down the line (think about executing in scripts, for example, where a user downloads a 5.1 package only to find out it has ended up in 5.3 instead). Supplying a Lua version is an explicit action, so we're doing the user a favour by letting them know about conflicts.

Reply:
right, I was thinking of how luarocks lets you specify the default lua version in a global config file.

Reply:
In the case of a global config then these shouldn't clash, you're right. I've been considering calling the future config value `preferred_lua_version` for this reason :)